framework/auth/views.py

```diff
@@ -230,6 +255,92 @@ def confirm_email_get(token, auth=None, **kwargs):
     ))
 
 
+@collect_auth
+def confirm_user_get(auth=None, **kwargs):
+    """Called at login to see if there are emails to add or users to merge.
+    methods: GET
+    """
+    # user = User.load(kwargs['uid'])
+    user = auth.user
+    verified_emails = []
+    for token in user.email_verifications:
+        if user.confirm_token(token):
+            if user.email_verifications[token]['confirmed']:
+                try:
+                    user_merge = User.find_one(Q('emails', 'eq', user.email_verifications[token]['email'].lower()))
+                except NoResultsFound:
+                    user_merge = False
+
+                verified_emails.append({'address': user.email_verifications[token]['email'],
+                                        'token': token,
+                                        'confirmed': user.email_verifications[token]['confirmed'],
+                                        'user_merge': user_merge.email if user_merge else False})
+            else:
+                #todo migrate to remove this hack
+                user.email_verifications[token]['confirmed'] = False
+    user.save()
+    return verified_emails
+
+
+@collect_auth
+def confirm_email_remove(auth=None, **kwargs):
+    """Called at login if user cancels their merge or email add.
+    methods: PUT
+    """
+    user = auth.user
+    confirmed_email = request.json.get('address')
+    email_verifications = user.email_verifications.copy()
```

Comment:
This [might not be doing](http://stackoverflow.com/questions/3975376/understanding-dict-copy-shallow-or-deep) what you think it's doing. If you just want to manipulate `user.email_verifications` (which seems to be the case), you probably don't need to make a (shallow or deep) copy at all. If you want to be able to make changes to the copied dict without affecting the original object in any way, you probably want to use `copy.deepcopy`.
```python
In [1]: a = {1: [1,2,3]}

In [2]: b = a.copy()

In [3]: b[1].append(4)

In [4]: a
Out[4]: {1: [1, 2, 3, 4]}

In [5]: b[2] = 0

In [6]: a
Out[6]: {1: [1, 2, 3, 4]}

In [7]: b
Out[7]: {1: [1, 2, 3, 4], 2: 0}

In [8]: b.pop(1)
Out[8]: [1, 2, 3, 4]

In [9]: a
Out[9]: {1: [1, 2, 3, 4]}

In [10]: b
Out[10]: {2: 0}
```